script.module.resolveurl/lib/resolveurl/plugins/vidtube.py

```diff
@@ -0,0 +1,37 @@
+"""
+    Plugin for ResolveURL
+    Copyright (C) 2023 bassemhelal18
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
+from resolveurl.lib import helpers
+
+
+class VidtubeResolver(ResolveGeneric):
```

Comment:
Classname (without the Resolver part) and name should match as it is used in cleaning settings. I will merge and change it, but please be aware of it when you create more reolvers